src/cargo/core/resolver/context.rs

```diff
@@ -318,8 +318,8 @@ fn build_requirements<'a, 'b: 'a>(
             uses_default_features: true,
             ..
         } => {
-            if s.features().get("default").is_some() {
-                reqs.require_feature("default")?;
+            if s.features().get(&InternedString::new("default")).is_some() {
```

Comment:
If `Borrow` is implemented for `InternedString` would it allow lookups with `&str` here perhaps?

Reply:
`impl Borrow<str> for InternedString` Hmmm. We are currently Hashing InternedString and str the same. So we could. I have a vain hope that we can find some way to have `O(1)` hashing. but we don't now. So we can always remove `Borrow` if we gain faster hashing.

Another peace is to make a singleton for each constant string we interne, it is mostly just "default".

Reply:
Yeah I think for now having `Borrow` may be the best bet in that it'll hopefully give us a nice ergonomic boost in cases like this and also avoid unnecessary temporary allocations in theory

Reply:
Added, and it is an ergonomic boost. It should also save one call to `hash` and 2 cash misses. `InternedString::new` only allocates if the value is unseen, so this was not really allocating in the hot path. But optimizers are odd so maybe removing the unused branch will help.